dm/master/scheduler/scheduler.go

```diff
@@ -1777,3 +1934,208 @@ func (s *Scheduler) SetWorkerClientForTest(name string, mockCli workerrpc.Client
 		s.workers[name].cli = mockCli
 	}
 }
+
+// nolint:dupl
+func (s *Scheduler) observeLoadTask(ctx context.Context, etcdCli *clientv3.Client, rev int64) error {
+	var wg sync.WaitGroup
+	for {
+		loadTaskCh := make(chan ha.LoadTask, 10)
+		loadTaskErrCh := make(chan error, 10)
+		wg.Add(1)
+		// use ctx1, cancel1 to make sure old watcher has been released
+		ctx1, cancel1 := context.WithCancel(ctx)
+		go func() {
+			defer func() {
+				close(loadTaskCh)
+				close(loadTaskErrCh)
+				wg.Done()
+			}()
+			ha.WatchLoadTask(ctx1, etcdCli, rev+1, loadTaskCh, loadTaskErrCh)
+		}()
+		err := s.handleLoadTask(ctx1, loadTaskCh, loadTaskErrCh)
+		cancel1()
+		wg.Wait()
+
+		if etcdutil.IsRetryableError(err) {
+			rev = 0
+			retryNum := 1
+			for rev == 0 {
+				select {
+				case <-ctx.Done():
+					return nil
+				case <-time.After(500 * time.Millisecond):
+					rev, err = s.syncLoadTaskFromEtcd(etcdCli)
+					if err != nil {
+						log.L().Error("resetLoadTask is failed, will retry later", zap.Error(err), zap.Int("retryNum", retryNum))
+					}
+				}
+				retryNum++
+			}
+		} else {
+			if err != nil {
+				log.L().Error("observeLoadTask is failed and will quit now", zap.Error(err))
+			} else {
+				log.L().Info("observeLoadTask will quit now")
+			}
+			return err
+		}
+	}
+}
+
+// RemoveLoadTask removes the loadtask by task.
+func (s *Scheduler) RemoveLoadTask(task string) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if !s.started {
+		return terror.ErrSchedulerNotStarted.Generate()
+	}
+	_, _, err := ha.DelLoadTaskByTask(s.etcdCli, task)
+	if err != nil {
+		return err
+	}
+	delete(s.loadTasks, task)
+	return nil
+}
+
+// getTransferWorkerAndSource tries to get transfer worker and source.
+// worker, source	This means a subtask finish load stage, often called by handleLoadTaskDel.
+// worker, ""		This means a free worker online, often called by tryBoundForWorker.
+// "", source		This means a unbounded source online, often called by tryBoundForSource.
+func (s *Scheduler) getTransferWorkerAndSource(worker, source string) (string, string) {
```

Comment:
what's the meaning of return value 😵

Reply:
Another worker and source. Given one worker and one source, get another worker and source to transfer.

Reply:
prefer to rename to "getNextLoadTaskTransfer" and update comment:

return (worker source) that is used by `transferWorkerAndSource`, to try to resolve a paused load task that the source can't be bound to the worker which has its dump files.